backend/tests/unit/dal/Services/TenantServiceTest.cs

```diff
@@ -0,0 +1,150 @@
+using System.Collections.Generic;
+using System.Diagnostics.CodeAnalysis;
+using Xunit;
+using Entity = Pims.Dal.Entities;
+
+namespace Pims.Dal.Test.Services
+{
+    [Trait("category", "unit")]
+    [Trait("category", "dal")]
+    [Trait("group", "tenants")]
+    [ExcludeFromCodeCoverage]
+    public class TenantServiceTest
+    {
+        #region Data
+        public static IEnumerable<object[]> TenantsWithId =>
+            new List<object[]>
+            {
+                new object[] { 0, 0 },
+                new object[] { 1, 2 }
+            };
+        #endregion
+
+        #region Tests
+        #region GetTenant
+        [Fact]
+        public void GetTenant()
+        {
+            // Arrange
+            var helper = new TestHelper();
+            var user = PrincipalHelper.CreateForPermission();
+
+            var init = helper.CreatePimsContext(user, true);
+            var tenant = EntityHelper.CreateTenant(1, "TEST", "Test Tenant");
+            init.AddAndSaveChanges(tenant);
+
+            var service = helper.CreateService<TenantService>(user);
+
+            // Act
+            var result = service.GetTenant(tenant.Code);
+
+            // Assert
+            Assert.NotNull(result);
+            Assert.IsAssignableFrom<Entity.Tenant>(result);
+            result.Should().Be(tenant);
+        }
+
+        [Fact]
+        public void GetTenant_ReturnNull()
+        {
+            // Arrange
+            var helper = new TestHelper();
+            var user = PrincipalHelper.CreateForPermission();
+
+            var init = helper.CreatePimsContext(user, true);
+            var tenant = EntityHelper.CreateTenant(1, "TEST", "Test Tenant");
+            init.AddAndSaveChanges(tenant);
+
+            var service = helper.CreateService<TenantService>(user);
+
+            // Act
+            var result = service.GetTenant("FAKE");
+
+            // Assert
+            result.Should().BeNull();
+        }
+        #endregion
+
+        #region UpdateTenant
+        [Fact]
+        public void UpdateTenant_Success()
+        {
+            // Arrange
+            var helper = new TestHelper();
+            var user = PrincipalHelper.CreateForPermission(Permissions.SystemAdmin);
+
+            var init = helper.CreatePimsContext(user, true);
+            var tenant = EntityHelper.CreateTenant(1, "TEST", "Test Tenant");
+            init.AddAndSaveChanges(tenant);
+
+            var service = helper.CreateService<TenantService>(user);
+
+            var updateTenant = EntityHelper.CreateTenant(1, "TEST", "New Name");
+            updateTenant.Description = "New Description";
+            updateTenant.Settings = "{\"id\":1}";
+
+            // Act
+            var result = service.UpdateTenant(updateTenant);
+
+            // Assert
+            Assert.NotNull(result);
+            Assert.IsAssignableFrom<Entity.Tenant>(result);
+            result.Should().Be(tenant);
+            result.Name.Should().Be(updateTenant.Name);
```

Comment:
this assertion library also supports deep equal comparisons which there are pros and cons of using.

Reply:
Here is is simply confirming the object returned is the same as the object from the database.